imu_sensor_broadcaster/test/test_imu_sensor_broadcaster.cpp

```diff
@@ -208,6 +209,154 @@ TEST_F(IMUSensorBroadcasterTest, SensorName_Publish_Success)
+  subscribe_and_get_message(imu_msg);
+
+  EXPECT_EQ(imu_msg.header.frame_id, frame_prefix + frame_id_);
+}
+
+TEST_F(IMUSensorBroadcasterTest, BlankPrefixNoNamespace)
+{
+  SetUpIMUBroadcaster();
+
+  std::string frame_prefix = "";
+
+  // set the params 'sensor_name' and 'frame_id'
+  imu_broadcaster_->get_node()->set_parameter({"sensor_name", sensor_name_});
+  imu_broadcaster_->get_node()->set_parameter({"frame_id", frame_id_});
+
+  imu_broadcaster_->get_node()->set_parameter({"tf_frame_prefix_enable", true});
+  imu_broadcaster_->get_node()->set_parameter({"tf_frame_prefix", frame_prefix});
+
+  ASSERT_EQ(imu_broadcaster_->on_configure(rclcpp_lifecycle::State()), NODE_SUCCESS);
+  ASSERT_EQ(imu_broadcaster_->on_activate(rclcpp_lifecycle::State()), NODE_SUCCESS);
+
+  sensor_msgs::msg::Imu imu_msg;
+  subscribe_and_get_message(imu_msg);
+
+  EXPECT_EQ(imu_msg.header.frame_id, frame_id_);
+}
+
+TEST_F(IMUSensorBroadcasterTest, NoPrefixWithNamespace)
+{
+  std::string test_namespace = "test_namespace";
+
+  SetUpIMUBroadcaster(test_namespace);
+
+  std::string frame_prefix = "test_prefix";
+
+  // set the params 'sensor_name' and 'frame_id'
+  imu_broadcaster_->get_node()->set_parameter({"sensor_name", sensor_name_});
+  imu_broadcaster_->get_node()->set_parameter({"frame_id", frame_id_});
+
+  imu_broadcaster_->get_node()->set_parameter({"tf_frame_prefix_enable", false});
+  imu_broadcaster_->get_node()->set_parameter({"tf_frame_prefix", frame_prefix});
+
+  ASSERT_EQ(imu_broadcaster_->on_configure(rclcpp_lifecycle::State()), NODE_SUCCESS);
+  ASSERT_EQ(imu_broadcaster_->on_activate(rclcpp_lifecycle::State()), NODE_SUCCESS);
+
+  sensor_msgs::msg::Imu imu_msg;
+  subscribe_and_get_message(imu_msg, test_namespace);
+
+  EXPECT_EQ(imu_msg.header.frame_id, frame_id_);
+}
+
+TEST_F(IMUSensorBroadcasterTest, PrefixWithNamespace)
+{
+  std::string test_namespace = "test_namespace";
+
+  SetUpIMUBroadcaster(test_namespace);
+
+  std::string frame_prefix = "test_prefix";
+
+  // set the params 'sensor_name' and 'frame_id'
+  imu_broadcaster_->get_node()->set_parameter(
+    rclcpp::Parameter("sensor_name", rclcpp::ParameterValue(sensor_name_)));
+  imu_broadcaster_->get_node()->set_parameter(
+    rclcpp::Parameter("sensor_name", rclcpp::ParameterValue(sensor_name_)));
+  imu_broadcaster_->get_node()->set_parameter({"frame_id", frame_id_});
+
+  imu_broadcaster_->get_node()->set_parameter(
+    rclcpp::Parameter("tf_frame_prefix_enable", rclcpp::ParameterValue(true)));
+  imu_broadcaster_->get_node()->set_parameter(
+    rclcpp::Parameter("tf_frame_prefix", rclcpp::ParameterValue(frame_prefix)));
+
+  ASSERT_EQ(imu_broadcaster_->on_configure(rclcpp_lifecycle::State()), NODE_SUCCESS);
+  ASSERT_EQ(imu_broadcaster_->on_activate(rclcpp_lifecycle::State()), NODE_SUCCESS);
+
+  sensor_msgs::msg::Imu imu_msg;
+  subscribe_and_get_message(imu_msg, test_namespace);
+
+  EXPECT_EQ(imu_msg.header.frame_id, frame_prefix + frame_id_);
```

Comment:
Shouldn't there be a slash in between?

Reply:
ok if you want it to be prefixed without a '/' at the end, why don't you parse it directly already appending it to the frame_id? This frame_id is only used for the published message. I don't see a point of adding a tf_prefix, if it is not namespaced.

https://github.com/ros-controls/ros2_controllers/blob/9e97c00496c239bf4961dc043a3f461aae9b23b8/imu_sensor_broadcaster/src/imu_sensor_broadcaster.cpp#L66

What do you think about this?